src/connectors/archive.js

```diff
@@ -41,39 +60,54 @@ Connector.isPlaying = () => {
 
 Connector.playerSelector = '#theatre-ia';
 
-Connector.trackArtSelector = '#theatre-ia center > img';
+Connector.trackArtSelector = [
+	// https://archive.org/details/AH013_sarin_sunday_-_the_lonely_hike
+	'#theatre-ia center > img',
 
-Connector.getArtistTrack = () => {
+	// https://archive.org/details/lp_everybody-knows-this-is-nowhere_neil-young-crazy-horse-robin-lane
+	'.album-cover img',
+];
+
+Connector.getTrackInfo = () => {
+	const artist = getArtists(artistSelectors);
+	let album = Util.getTextFromSelectors(albumSelector);
 	let track = Util.getTextFromSelectors(trackSelectors);
-	const artist = Util.getTextFromSelectors(artistSelectors);
 
-	const trackParts = track.split('-').map((item) => {
-		return item.trim();
-	});
+	if (track) {
+		// Second item could be an artist name or track duration
+		const [firstItem, secondItem] = track
+			.split('-')
+			.map((item) => item.trim());
 
-	if (trackParts.length === 3 && trackParts[0] === artist) {
-		track = trackParts[2];
+		if (artist.includes(secondItem)) {
+			track = firstItem;
+		}
+	} else {
+		track = album;
+		album = null;
 	}
 
-	return { artist, track };
+	return { artist, track, album };
 };
 
-Connector.albumSelector = '.thats-left > h1 [itemprop=name]';
-
-// Example: https://archive.org/details/AH003_corwin_trails_-_corwin_trails
-function hasAllTracksNumericPrefix(trackSelector) {
-	const tracks = document.querySelectorAll(trackSelector);
+function hasAllTracksNumericPrefix() {
+	const tracks = document.querySelectorAll(tracksSelector);
 	if (tracks.length === 0) {
 		return false;
 	}
 
 	let hasAllTracksNumericPrefix = true;
 	for (const track of tracks) {
-		if (!numericTrackRegex.test(track.textContent)) {
+		if (!numericTrackRegex.test(track.textContent.trim())) {
 			hasAllTracksNumericPrefix = false;
 			break;
 		}
 	}
 
 	return hasAllTracksNumericPrefix;
 }
+
+function getArtists(selectors) {
```

Comment:
It gets text from all nodes (artist nodes) and returns a string of text values joined with comma.

For example
```
<a href="..." class="artist">Artist 1</a>
<a href="..." class="artist">Artist 2</a>
```

`getArtists('.artist')` will return "Artist 1, Artist 2" which is more common (vs raw "Artist 1; Artist 2" joined with semicolon, which is provided by archive.org).

For now it's copy-pasted from other connectors; though it should be extracted as an utility function.

In future we can send an array of artists to the background script for easier manipulation instead of sending a string.

Reply:
Ahh so the background handles the extraction of the artist from that CSV?